src/Datadog.Trace.ClrProfiler.Managed/Integrations/AdoNetIntegration.cs

```diff
@@ -4,66 +4,117 @@
             TargetMinimumVersion = Major4,
             TargetMaximumVersion = Major4)]
-        public static object ExecuteDbDataReader(
-            object @this,
-            int behavior,
+        public static object ExecuteReader(
+            object command,
             int opCode,
             int mdToken,
             long moduleVersionPtr)
         {
-            if (@this == null)
+            Func<DbCommand, DbDataReader> instrumentedMethod;
+
+            try
             {
-                throw new ArgumentNullException(nameof(@this));
+                instrumentedMethod =
+                    MethodBuilder<Func<DbCommand, DbDataReader>>
+                       .Start(moduleVersionPtr, mdToken, opCode, AdoNetConstants.MethodNames.ExecuteReader)
+                       .WithConcreteType(typeof(DbCommand))
+                       .WithNamespaceAndNameFilters(DbDataReaderTypeName)
+                       .Build();
+            }
+            catch (Exception ex)
+            {
+                Log.ErrorRetrievingMethod(
+                    exception: ex,
+                    moduleVersionPointer: moduleVersionPtr,
+                    mdToken: mdToken,
+                    opCode: opCode,
+                    instrumentedType: DbCommandTypeName,
+                    methodName: nameof(ExecuteReader),
+                    instanceType: command.GetType().AssemblyQualifiedName);
+                throw;
             }
 
-            Func<object, CommandBehavior, object> instrumentedMethod = null;
+            var dbCommand = command as DbCommand;
+
+            using (var scope = ScopeFactory.CreateDbCommandScope(Tracer.Instance, dbCommand, IntegrationName))
+            {
+                try
+                {
+                    return instrumentedMethod(dbCommand);
+                }
+                catch (Exception ex)
+                {
+                    scope?.Span.SetException(ex);
+                    throw;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Instrumentation wrapper for <see cref="DbCommand.ExecuteDbDataReader"/>.
+        /// </summary>
+        /// <param name="command">The object referenced "this" in the instrumented method.</param>
+        /// <param name="behavior">The <see cref="CommandBehavior"/> value used in the original method call.</param>
+        /// <param name="opCode">The OpCode used in the original method call.</param>
+        /// <param name="mdToken">The mdToken of the original method call.</param>
+        /// <param name="moduleVersionPtr">A pointer to the module version GUID.</param>
+        /// <returns>The value returned by the instrumented method.</returns>
+        [InterceptMethod(
+            TargetAssemblies = new[] { AdoNetConstants.AssemblyNames.SystemData, AdoNetConstants.AssemblyNames.SystemDataCommon },
+            TargetType = DbCommandTypeName,
```

Comment:
As noted above in `integrations.json`, can you include the `TargetMethod = "ExecuteReader"` here?

Reply:
Fixed in a1a7fd28ab97bf73f89a55e44cdc2b73394aca20.